CPS-0010/README.md

```diff
@@ -0,0 +1,276 @@
+Although, dApps should be able to request additional functionalities on connection if they wish.
+
+This does not mean that all connections must support cryptographic operations.
+
+#### Types of Wallet by Query Layer configuration
+
+All possible wallet standards can be divided into three groups based on a single criterion: what data they provide to dApps via the API endpoint.
+
+This division is important because each of the groups allows for different dApp architectures.
+
+Note that the groups are nested: every wallet is a no-data wallet and every full-data wallet is also an own-data wallet if we look at the dApp architectures they enable.
+
+##### No-data wallets
+
+No-data wallets do not provide data queries and are only concerned with cryptographic operations (e.g. hardware wallets).
+All management of UTxOs is placed on dApps side.
+No-data wallets can use multiple addresses, but they do not allow to query for available UTxOs and do not indicate which of the addresses are actually used on-chain.
+
+These wallets may even rely on dApps to derive addresses, by sharing the root public key.
+
+
+![diagram showing wallet and dApp architecture for no-data wallets](./no-data-wallet.drawio.png)
+
+This design ensures that wallets can function without a query layer and thus without runtime infrastructure needed.
+
+##### Own-data wallets
+
+Own-data wallets provide chain queries related to users' own data and wallet state (users' addresses and utxos).
+CIP-30 falls into this category.
+
+![diagram showing possible wallet and dApp architecture for own-data wallets](./own-data-wallet.drawio.png)
+
+This architecture has two major issues:
+
+- **Unclear separation of concerns** - while wallets provide certain functions for dApp developers, they have no means to enforce their use, and naturally some developers opt to using their own query layers when they see fit, with consequences that are hard to predict. For example, the wallet may not allow signing a transaction that consumes an UTxO it considers "locked".
+
+- **Two sources of truth problem** - on Cardano, every running node has its own opinion on the set of currently unspent transaction outputs. Only [eventual consistency](https://en.wikipedia.org/wiki/Eventual_consistency) is guaranteed. Any dApp that interacts with an own-data wallet has to deal with the inconsistency between local `cardano-node`-based query layer and light wallet query layer, especially when dApp workflow involves sending multiple interactions with the wallet in quick succession. For example, a wallet may refuse to sign a transaction containing an UTxO it does not (yet) know about.
+
+Even if we enforce clear separation of concerns in the standards, e.g. by stating that sending a transaction bypassing the wallet is not allowed, the problem of two sources of truth will remain.
+
+##### Full-data wallets
+
+Full-data wallets allow to query blockchain data outside of user's scope (i.e. anything not covered by own-data wallets).
```

Comment:
```suggestion
Full-data wallets allow querying blockchain data outside of user's scope (i.e. anything not covered by own-data wallets).
```
(grammar: unlike other languages, in English the verb tense here would be participle rather than infinitive)